clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1724,6 +1738,21 @@ private void subscribeInternal(Pattern pattern, Optional<ConsumerRebalanceListen
         }
     }
 
+    private void subscribeInternal(SubscriptionPattern pattern, Optional<ConsumerRebalanceListener> listener) {
+        acquireAndEnsureOpen();
+        try {
+            maybeThrowInvalidGroupIdException();
+            if (pattern == null || pattern.pattern().isEmpty())
+                throw new IllegalArgumentException("Topic pattern to subscribe to cannot be " + (pattern == null ?
+                    "null" : "empty"));
+            throwIfNoAssignorsConfigured();
+            log.info("Subscribed to pattern: '{}'", pattern);
```

Comment:
This will probably not work since you did not add a `toString()` method to `SubscriptionPattern`. You should either add `toString()` to `SubscriptionPattern` or change this line to
```suggestion
            log.info("Subscribed to pattern: '{}'", pattern.pattern());
```